geode-core/src/main/java/org/apache/geode/distributed/internal/direct/DirectChannel.java

```diff
@@ -458,8 +472,14 @@ private ConnectExceptions getConnections(Membership mgr, DistributionMessage msg
           if (ackTimeout > 0) {
             startTime = System.currentTimeMillis();
           }
-          Connection con = conduit.getConnection(destination, preserveOrder, retry, startTime,
-              ackTimeout, ackSDTimeout);
+          Connection con;
```

Comment:
Can this be `final`?
Perhaps this if block should be extracted into its own method for clarity.
I wouldn't mind a rename of `con` to `connection` while we are here.